internal/pwru/output.go

```diff
@@ -247,10 +248,20 @@ func getStackData(event *Event, o *output) (stackData string) {
 
 func getSkbData(event *Event, o *output) (skbData string) {
 	id := uint32(event.PrintSkbId)
-	if str, err := o.printSkbMap.LookupBytes(&id); err == nil {
-		skbData = string(str)
+
+	b, err := o.printSkbMap.LookupBytes(&id)
+	if err != nil {
+		return ""
+	}
+
+	length := binary.NativeEndian.Uint32(b[:4])
+
+	// Bounds check
+	if int(length+4) > len(b) {
+		return ""
 	}
-	return skbData
+
+	return "\n" + string(b[4:4+length])
```

Comment:
Didn't you mean `return string(b[4:4+length]) + "\n"` instead?

Reply:
No, this is intended. Currently, we start to print skb right after the function name

```
0xffff88886f430d00      6 [udp_main:913171]             kfree_skbmem(struct sk_buff){
 (union){
  (struct){
   (union){
    .dev = (struct net_device *)0x000000000e329f1c,
    .dev_scratch = (long unsigned int)18446612686734655488,
   },
```

This makes it a bit cleaner.

```
0xffff88886f430d00      6 [udp_main:913171]             kfree_skbmem
(struct sk_buff){
 (union){
  (struct){
   (union){
    .dev = (struct net_device *)0x000000000e329f1c,
    .dev_scratch = (long unsigned int)18446612686734655488,
   },
```